homeassistant/components/here_weather/config_flow.py

```diff
@@ -0,0 +1,120 @@
+"""Config flow for here_weather integration."""
+from __future__ import annotations
+
+import logging
+
+import herepy
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_NAME,
+    CONF_SCAN_INTERVAL,
+)
+from homeassistant.core import HomeAssistant, callback
+import homeassistant.helpers.config_validation as cv
+
+from .const import DEFAULT_MODE, DEFAULT_SCAN_INTERVAL, DOMAIN, HERE_API_KEYS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_validate_user_input(hass: HomeAssistant, user_input: dict) -> None:
+    """Validate the user_input containing coordinates."""
+    here_client = herepy.DestinationWeatherApi(user_input[CONF_API_KEY])
+    await hass.async_add_executor_job(
+        here_client.weather_for_coordinates,
+        user_input[CONF_LATITUDE],
+        user_input[CONF_LONGITUDE],
+        herepy.WeatherProductType[DEFAULT_MODE],
+    )
+
+
+class HereWeatherConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for here_weather."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry: ConfigEntry):
+        """Get the options flow for this handler."""
+        return HereWeatherOptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            try:
+                await async_validate_user_input(self.hass, user_input)
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME], data=user_input
+                )
+            except herepy.InvalidRequestError:
+                errors["base"] = "invalid_request"
+            except herepy.UnauthorizedError:
+                errors["base"] = "unauthorized"
+        return self.async_show_form(
+            step_id="user",
+            data_schema=self._get_schema(user_input),
+            errors=errors,
+        )
+
+    def _get_schema(self, user_input: dict | None) -> vol.Schema:
+        known_api_key = None
+        if HERE_API_KEYS in self.hass.data:
+            known_api_key = self.hass.data[HERE_API_KEYS][0]
```

Comment:
Just iterate the existing config entries and get the keys that way instead.

https://github.com/home-assistant/core/blob/596179d1800ddc11b108602d5c073af850744d36/homeassistant/config_entries.py#L1219-L1226

Why are we only expecting a single existing entry?

Reply:
Fixed in https://github.com/home-assistant/core/pull/28910/commits/34104ed8240aef0e98e90517905aadf757958d97